scripts/skills/actives/rf_dynamic_duo_shuffle_skill.nut

```diff
@@ -47,20 +45,30 @@ this.rf_dynamic_duo_shuffle_skill <- ::inherit("scripts/skills/skill", {
 		if (this.getContainer().getActor().getCurrentProperties().IsRooted)
 		{
 			ret.push({
-				id = 9,
+				id = 20,
 				type = "text",
 				icon = "ui/tooltips/warning.png",
-				text = "[color=" + ::Const.UI.Color.NegativeValue + "]Cannot be used while rooted[/color]"
+				text = ::MSU.Text.colorRed("Cannot be used while rooted")
 			});
 		}
 
 		if (this.m.IsSpent)
 		{
 			ret.push({
-				id = 9,
+				id = 21,
 				type = "text",
 				icon = "ui/tooltips/warning.png",
-				text = "[color=" + ::Const.UI.Color.NegativeValue + "]Cannot be used more than once per turn[/color]"
+				text = ::Reforged.Mod.Tooltips.parseString(::MSU.Text.colorRed("Cannot be used more than once per [turn|Concept.Turn]"))
+			});
+		}
+
+		if (::MSU.isNull(this.m.DynamicDuoPerk.getPartner()) || !this.m.DynamicDuoPerk.getPartner().isPlacedOnMap())
+		{
+			ret.push({
+				id = 22,
+				type = "text",
+				icon = "ui/tooltips/warning.png",
+				text = ::MSU.Text.colorRed("Can only be used when next to your partner")
```

Comment:
The warning here only barely fits with the conditions you test for.
I feel like the cleanest is to have an `if` into an `else if` where you test for both conditions above separately and then write a much more precise error message depending on the case.

And if you really find this warning about the partner needing to be adjacent, important. Then you should introduce a new third condition for this case to check for.

Reply:
I've rearranged the tooltip to be clearer about the blocking conditions. Please take another look.